File: 0776-n-ary-tree-postorder-traversal/0776-n-ary-tree-postorder-traversal.py
"""
# Definition for a Node.
class Node(object):
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

class Solution(object):
    def postorder(self, root):
        """
        :type root: Node
        :rtype: List[int]
        """
        vec = []
        def recurse(node):
            if node is None:
                return
            for child in node.children:
                recurse(child)
            vec.append(node.val)
        recurse(root)
        return vec
        

File: 0776-n-ary-tree-postorder-traversal/0776-n-ary-tree-postorder-traversal.cpp
/*
// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val) {
        val = _val;
    }

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
public:
    void recurse(Node* node, vector<int>& vec){
        if (node == nullptr) return;
        for (auto child : node->children){
            recurse(child, vec);
        }
        vec.push_back(node->val);        
    }
    
    vector<int> postorder(Node* root) {
        vector<int> result;
        recurse(root, result);
        return result;
    }
};